examples/FeatureFlagDemo/AssignUserMiddleware.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace FeatureFlagDemo
+{
+    /// <summary>
+    /// This is a contrived middleware used to assign user information to the request based off of parameters passed in via the request's query string parameters.
+    /// Typically an application will authenticate a user and populate user information during that process.
+    /// 
+    /// To assign a user use the following query string structure "?username=JohnDoe&groups=MyGroup1,MyGroup2"
+    /// </summary>
+    public class AssignUserMiddleware
```

Comment:
Normally, this will be done as part of authentication in a real-world app. Can we mimic that in the example too?

Reply:
Updated this to use an authentication handler and a demo authentication scheme, `QueryString`